skunkworks_crow/src/main/java/org/odk/share/activities/SendActivity.java

```diff
@@ -423,49 +360,36 @@ protected Dialog onCreateDialog(int id) {
                 progressDialog.setIndeterminate(true);
                 progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
                 progressDialog.setCancelable(false);
-                progressDialog.setButton(DialogInterface.BUTTON_NEGATIVE, getString(R.string.cancel),
-                        new DialogInterface.OnClickListener() {
-                            @Override
-                            public void onClick(DialogInterface dialog, int which) {
-                                senderService.cancel();
-                                dialog.dismiss();
-                                finish();
-                            }
-                        });
+                progressDialog.setButton(DialogInterface.BUTTON_NEGATIVE, getString(R.string.cancel), (dialog, which) -> {
+                    senderService.cancel();
+                    dialog.dismiss();
+                    finish();
+                });
                 return progressDialog;
         }
 
         return null;
     }
 
     private void createAlertDialog(String title, String message) {
-        AlertDialog alertDialog = new AlertDialog.Builder(this).create();
-        alertDialog.setTitle(title);
-        alertDialog.setMessage(message);
-        DialogInterface.OnClickListener quitListener = new DialogInterface.OnClickListener() {
-            @Override
-            public void onClick(DialogInterface dialog, int i) {
-                switch (i) {
-                    case DialogInterface.BUTTON_POSITIVE:
-                        stopHotspot();
-                        finish();
-                        break;
-                }
-            }
-        };
-        alertDialog.setCancelable(false);
-        alertDialog.setButton(DialogInterface.BUTTON_POSITIVE, getString(R.string.ok), quitListener);
-        alertDialog.show();
+        new AlertDialog.Builder(this)
```

Comment:
So many similar dialogs, can we building something like `DailogFactory.java` which has different static methods can return dialogs?